src/AstLogStatement.h

```diff
@@ -0,0 +1,229 @@
+        const char* messageType = "@t-recursive-relation";
+        std::stringstream line;
+        if (AstLogStatement::getLogFileType() == "json") {
+            line << "{";
+            line << "\"msg-type\": \"" << messageType << "\", ";
+            line << "\"relation\": \"" << relationName << "\", ";
+            line << "\"src-locator\": \"" << srcLocation << "\", ";
+            line << "\"time\": ";
+        } else {
+            line << messageType << ";" << relationName << ";" << srcLocation << ";";
+        }
+        return line.str();
+    }
+
+    static const std::string nRecursiveRelation(
+            const std::string& relationName, const AstSrcLocation& srcLocation) {
+        const char* messageType = "@n-recursive-relation";
+        std::stringstream line;
+        if (AstLogStatement::getLogFileType() == "json") {
+            line << "{";
+            line << "\"msg-type\": \"" << messageType << "\", ";
+            line << "\"relation\": \"" << relationName << "\", ";
+            line << "\"src-locator\": \"" << srcLocation << "\", ";
+            line << "\"tuples\": ";
+        } else {
+            line << messageType << ";" << relationName << ";" << srcLocation << ";";
+        }
+        return line.str();
+    }
+
+    static const std::string cRecursiveRelation(
+            const std::string& relationName, const AstSrcLocation& srcLocation) {
+        const char* messageType = "@c-recursive-relation";
+        std::stringstream line;
+        if (AstLogStatement::getLogFileType() == "json") {
+            line << "{";
+            line << "\"msg-type\": \"" << messageType << "\", ";
+            line << "\"relation\": \"" << relationName << "\", ";
+            line << "\"src-locator\": \"" << srcLocation << "\", ";
+            line << "\"time\": ";
+        } else {
+            line << messageType << ";" << relationName << ";" << srcLocation << ";";
+        }
+        return line.str();
+    }
+
+    static const std::string pProofCounter(const std::string& relationName, const AstSrcLocation& srcLocation,
+            const std::string& datalogText) {
+        // TODO: the profiler should be modified to use this type of log message, as currently these messages
+        // are ignored
+        const char* messageType = "#p-proof-counter";
+        std::stringstream line;
+        if (AstLogStatement::getLogFileType() == "json") {
+            line << "{";
+            line << "\"msg-type\": \"" << messageType << "\", ";
+            line << "\"relation\": \"" << relationName << "\", ";
+            line << "\"src-locator\": \"" << srcLocation << "\", ";
+            line << "\"rule\": \"" << datalogText << "\", ";
+            line << "\"failed-proofs\": ";
+        } else {
+            line << messageType << ";" << relationName << ";" << srcLocation << ";" << datalogText << ";";
+            // TODO: the additional semicolon is added to maintain backwards compatibility and should
```

Comment:
TODOs should be in the form
`// TODO (username): description`
or
`// TODO (#issue number): description`
In this case, the issue number for the new profiler issue.